test/unit/db/repos/blocks.js

```diff
@@ -0,0 +1,885 @@
+	'b_height',
+	'b_id',
+	'b_numberOfTransactions',
+	'b_payloadHash',
+	'b_payloadLength',
+	'b_previousBlock',
+	'b_reward',
+	'b_timestamp',
+	'b_totalAmount',
+	'b_totalFee',
+	'b_version',
+	'd_username',
+	'dapp_category',
+	'dapp_description',
+	'dapp_icon',
+	'dapp_link',
+	'dapp_name',
+	'dapp_tags',
+	'dapp_type',
+	'in_dappId',
+	'm_keysgroup',
+	'm_lifetime',
+	'm_min',
+	'ot_dappId',
+	'ot_outTransactionId',
+	's_publicKey',
+	't_amount',
+	't_fee',
+	't_id',
+	't_recipientId',
+	't_requesterPublicKey',
+	't_rowId',
+	't_senderId',
+	't_senderPublicKey',
+	't_signSignature',
+	't_signature',
+	't_signatures',
+	't_timestamp',
+	't_type',
+	'tf_data',
+	'v_votes',
+];
+
+describe('db', () => {
+	before(done => {
+		dbSandbox = new DBSandbox(__testContext.config.db, 'lisk_test_db_blocks');
+
+		dbSandbox.create((err, __db) => {
+			db = __db;
+
+			done(err);
+		});
+	});
+
+	after(done => {
+		dbSandbox.destroy();
+		done();
+	});
+
+	beforeEach(done => {
+		seeder
+			.seed(db)
+			.then(() => done(null))
+			.catch(done);
+	});
+
+	afterEach(done => {
+		sinonSandbox.restore();
+		seeder
+			.reset(db)
+			.then(() => done(null))
+			.catch(done);
+	});
+
+	it('should initialize db.blocks repo', done => {
+		expect(db.blocks).to.be.not.null;
+		done();
+	});
+
+	describe('AccountsRepository', () => {
+		describe('constructor()', () => {
+			it('should assign param and data members properly', done => {
+				expect(db.blocks.db).to.be.eql(db);
+				expect(db.blocks.pgp).to.be.eql(db.$config.pgp);
+				expect(db.blocks.dbTable).to.be.eql('blocks');
+
+				expect(db.blocks.dbFields).to.be.an('array');
+				expect(db.blocks.dbFields).to.have.lengthOf(13);
+
+				expect(db.blocks.sortFields).to.be.an('array');
+				expect(db.blocks.sortFields).to.have.lengthOf(9);
+
+				// TODO: Need to explore way to rewire an initialized module to fetch private members
+				// expect(db.blocks.cs).to.be.an('object');
+				// expect(db.blocks.cs).to.be.not.empty;
+				// expect(db.blocks.cs).to.have.all.keys(['insert']);
+
+				done();
+			});
+		});
+
+		describe('getGenesisBlock()', () => {
+			it('should use the correct SQL file with no parameters', function*() {
+				sinonSandbox.spy(db, 'any');
+				yield db.blocks.getGenesisBlock();
+
+				expect(db.any.firstCall.args[0]).to.eql(sql.getGenesisBlock);
+				expect(db.any.firstCall.args[1]).to.eql(undefined);
+				return expect(db.any).to.be.calledOnce;
+			});
+
+			it('should return genesis block with correct values', function*() {
+				const genesisBlock = yield db.blocks.getGenesisBlock();
+
+				expect(genesisBlock).to.not.be.empty;
+				expect(genesisBlock).to.have.lengthOf(1);
+				expect(genesisBlock[0]).to.have.all.keys(
+					'id',
+					'payloadHash',
+					'blockSignature'
+				);
+
+				const payloadHash = genesisBlock[0].payloadHash.toString('hex');
+
+				return expect(payloadHash).to.be.eql(__testContext.config.nethash);
+			});
+		});
+
+		describe('getGenesisBlockId()', () => {
```

Comment:
Consistency - `getGenesisBlockId` or `getGenesisBlockId(id)` (I like first more)

Reply:
I used `getGenesisBlockId()` just to show in tests results that its a function that is tested. To differentiate from testing some data attribute.